membership-attribute-service/app/models/AccountDetails.scala

```diff
@@ -60,16 +63,35 @@ object AccountDetails {
         case _ => Json.obj()
       }
 
-      val alertText = membershipAlertText match {
-        case Some(text) => Json.obj("alertText" -> text)
-        case None => Json.obj()
-      }
+      def planIsCurrent(plan: SubscriptionPlan.AnyPlan) = (plan.start == now || plan.start.isBefore(now)) && (plan match {
+        case paidPlan: SubscriptionPlan.Paid => paidPlan.end.isAfter(now)
+        case _ => true
+      })
+
+      def jsonifyPlan(plan: SubscriptionPlan.AnyPlan) = Json.obj(
+        "productRatePlanId" -> plan.productRatePlanId.get, // consider exposing hash of this to avoid exposing internal IDs
+        "productName" -> plan.productName,
+        "name" -> plan.name,
```

Comment:
Same point as above really - I would prefer it if the rate plan names from the Zuora catalog didn't get passed all the way through to a client (appreciate that this is already being leaked in the current responses though)

Reply:
For example, we probably want clients to treat all the different versions of Guardian Weekly that we have in the product catalog the same, so it would be nicer if we defined a clean type here and returned that instead of the exact string from Zuora.

Reply:
@jacobwinch I'll have a go at doing a cleaner function here

Reply:
done in https://github.com/guardian/members-data-api/pull/371/commits/89f6ea4f6f0f6f30eab0866210ff1739683f72b3 (with accompanying changes in https://github.com/guardian/manage-frontend/pull/196/commits/a0bb42cf24f8074eb3852161f9b094dce6210606)